src/main/java/com/wavefront/opentracing/WavefrontSpanBuilder.java

```diff
@@ -148,7 +150,35 @@ private WavefrontSpanContext createSpanContext() {
     WavefrontSpanContext traceCtx = traceAncestry();
     UUID traceId = (traceCtx == null) ? UUID.randomUUID() : traceCtx.getTraceId();
     Boolean sampling = (traceCtx == null) ? null : traceCtx.getSamplingDecision();
-    return new WavefrontSpanContext(traceId, spanId, null, sampling);
+    return new WavefrontSpanContext(traceId, spanId, getBaggage(), sampling);
+  }
+
+  @Nullable
+  private Map<String, String> getBaggage() {
+    return addItems(follows, addItems(parents, null));
+  }
+
+  /**
+   * Gets a map containing baggage items of all the given references.
+   *
+   * @param references the list of references to process
+   * @param baggage the map to add items to, can be null in which case a new map is created/returned
+   * @return the map containing baggage items from all references
+   */
+  @Nullable
+  private Map<String, String> addItems(List<Reference> references, Map<String, String> baggage) {
```

Comment:
Seems baggage parameter can be null. Indicate the same for parameter with `@Nullable`. Also the input parameter is being mutated and being returned. Though its utility method, would prefer to avoid that.

Reply:
Done this way for memory efficiency. 99% of spans will not have any baggage items. Wanted to minimize unnecessary allocations and duplicating code. It's only mutated if input is null.